flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/HybridSource.java

```diff
@@ -92,13 +90,13 @@
 
     private final List<SourceListEntry> sources;
     // sources are populated per subtask at switch time
-    private final Map<Integer, Source> switchedSources;
+    private final HybridSourceSplitSerializer.SwitchedSources switchedSources;
```

Comment:
why is `SwitchedSources` nested inside the `HybridSourceSplitSerializer`?

Reply:
did I miss sth? I didn't see `switchedSources` used anywhere?

Reply:
Yes please remove. `switchedSources` acted like a shared cache which is now not necessary anymore. (Not sure how I missed that in the initial review, I guess I was too focused on API.)

This should now just be a field in enumerator/reader that caches the sources.

Reply:
That came in after moving away from the fixed source sequence that originally both, enumerator and serializer had access to. They still needed access to the underlying serializer and therefore to the source that provided that serializer. Now the the serializers are decoupled, this is no longer needed. I just missed that in the refactor, thanks  @stevenzwu for catching it.